lib/rules.js

```diff
@@ -184,9 +185,60 @@ export const ABSOLUTE_VALUE = defineRuleString('|-#a|', '#a')
 
 // MULTIPLYING POLYNOMIALS
 
+// e.g. 6x y z -> 6 (x^1y^1z^1) (where x, y, z are
+// separate identifiers)
+
+export const ADD_EXPONENT_OF_ONE_HELPER = defineRule(
+    (node) => {
+        const hasConstantVariable = query.isMul(node) && node.args.some(query.isIdentifier)
+
+        return (hasConstantVariable) ? {node} : null
+    },
+
+    (node) => {
+        const result = build.applyNode(
+            'mul',
+            node.args.map(arg => {
+                if (query.isIdentifier(arg)){
+                    return build.applyNode('pow', [arg, build.numberNode(1)])
+                }
+                return arg
+            }), {implicit: node.implicit}
+        )
+        return result
+    }
+)
+
 // e.g. x^2 * x -> x^2 * x^1
-// export const ADD_EXPONENT_OF_ONE = ...
+export const ADD_EXPONENT_OF_ONE = defineRule(
+    (node) => {
+        let isMulOfPolynomials = false
+
+        if (query.isMul(node)) {
+            const {constants, coefficientMap} = getCoefficientsAndConstants(node)
+            isMulOfPolynomials = Object.keys(coefficientMap).length > 1
+                || Object.keys(coefficientMap)
+                .some(key => coefficientMap[key].length > 1)
```

Comment:
`getVariableFactors` doesn't really help in the case of :
`2x + 3x^2 * x y z => 2 x + 3 x^2 * x^1 y^1 z^1`, you will also get the `x` from `2x`